CollapsibleExample/MHPackagedFilter.m

```diff
@@ -0,0 +1,64 @@
+//
+//  MHPackagedFilter.m
+//  CollapsibleExample
+//
+//  Created by Lizzy Randall on 6/10/15.
+//  Copyright (c) 2015 Lizzy Randall. All rights reserved.
+//
+
+#import "MHPackagedFilter.h"
+
+@interface MHPackagedFilter()
+
+@property (nonatomic, strong) NSMutableArray *filterKeyValuePairs;
+@property (nonatomic) BOOL hierarchy;
+
+@end
+
+@implementation MHPackagedFilter
+
+- (instancetype)initWithRootKey:(NSString*)rootKey rootValue:(NSString*)rootValue hierarchy:(BOOL)hierarchy{
+    
+    self = [self initWithKey:rootKey value:rootValue];
+    self.filterKeyValuePairs = [[NSMutableArray alloc] init];
+    self.hierarchy = hierarchy;
+    
+    return self;
+}
+
+- (void)addFilterWithKey:(NSString*)key value:(NSString*)value{
+    
+    MHKeyValuePair *keyValue = [[MHKeyValuePair alloc] initWithKey:key value:value];
+    [self.filterKeyValuePairs addObject:keyValue];
```

Comment:
You can ignore this line given your above comment. Although you should know that you can add literal notation to custom classes (you can read about it here http://nshipster.com/object-subscripting/) However, I wouldn't recommend it for this class. I think having it written out long form shows that it is different from a regular dictionary.